pkg/prometheus/statefulset.go

```diff
@@ -932,14 +932,23 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *operator.Config, shard in
 		return nil, errors.Wrap(err, "failed to merge init containers spec")
 	}
 
+	if len(p.Spec.AdditionalArgs) > 0 {
+		if err := verifyAdditionalArgs(promArgs, p.Spec.AdditionalArgs); err != nil {
+			return nil, err
+		}
+		promArgs = append(promArgs, p.Spec.AdditionalArgs...)
+	}
+
+	containerArgs := createContainerArgs(promArgs)
```

Comment:
We can have a single function to merge and return the final args list.

```suggestion
	containerArgs, err := buildArgs(promArgs, p.Spec.AdditionalArgs...)
	if err != nil {
		return nil, err
	}
	...
}
	
func buildArgs(args []monitoringv1.Argument, additionalArgs monitoringv1.Argument...) ([]string, error) {
...
}
```